From 81d5d2c8328c48fcb0e0a6b8976a35a837bbef5b Mon Sep 17 00:00:00 2001
From: Marcin Niestroj <m.niestroj@emb.dev>
Date: Thu, 14 Aug 2025 12:30:36 +0200
Subject: [PATCH] nsi: fcntl: add O_CREAT and O_TRUNC

Support conversion of those flags between mid and zephyr/host.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>
---
 scripts/native_simulator/common/src/include/nsi_fcntl.h | 2 ++
 scripts/native_simulator/common/src/nsi_fcntl.c         | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/scripts/native_simulator/common/src/include/nsi_fcntl.h b/scripts/native_simulator/common/src/include/nsi_fcntl.h
index 891ce8a6632..63be9bda944 100644
--- a/scripts/native_simulator/common/src/include/nsi_fcntl.h
+++ b/scripts/native_simulator/common/src/include/nsi_fcntl.h
@@ -11,6 +11,8 @@
 #define NSI_FCNTL_MID_O_WRONLY		01
 #define NSI_FCNTL_MID_O_RDWR		02
 
+#define NSI_FCNTL_MID_O_CREAT		0x0100
+#define NSI_FCNTL_MID_O_TRUNC		0x0200
 #define NSI_FCNTL_MID_O_APPEND		0x0400
 #define NSI_FCNTL_MID_O_EXCL		0x0800
 #define NSI_FCNTL_MID_O_NONBLOCK	0x4000
diff --git a/scripts/native_simulator/common/src/nsi_fcntl.c b/scripts/native_simulator/common/src/nsi_fcntl.c
index d1fb298a532..487824e9f4c 100644
--- a/scripts/native_simulator/common/src/nsi_fcntl.c
+++ b/scripts/native_simulator/common/src/nsi_fcntl.c
@@ -44,6 +44,8 @@ static int nsi_fcntl_to_mid_(int flags, bool strict)
 	TO_MID(O_WRONLY);
 	TO_MID(O_RDWR);
 
+	TO_MID(O_CREAT);
+	TO_MID(O_TRUNC);
 	TO_MID(O_APPEND);
 	TO_MID(O_EXCL);
 	TO_MID(O_NONBLOCK);
@@ -81,6 +83,8 @@ int nsi_fcntl_from_mid(int flags_mid)
 	FROM_MID(O_WRONLY);
 	FROM_MID(O_RDWR);
 
+	FROM_MID(O_CREAT);
+	FROM_MID(O_TRUNC);
 	FROM_MID(O_APPEND);
 	FROM_MID(O_EXCL);
 	FROM_MID(O_NONBLOCK);
-- 
2.52.0

